### ISSUE TITLE
fix/upgrade integration tests

### ISSUE THREAD

**Issue by DavidGOrtega** [iterative/cml#538]
- tests sometimes pass despite introducing breaking changes
  + The main reason was having introduced a gyp lib that needed sudo permissions breaking the cml service inside the CML aws image managed by our terraform provider
- should use external repo & `send-check` to
  + avoid spamming CML with comments
  + avoid too many token permissions

We need:
 - Integration tests that take the whole picture 
    - testbot should do the trick however the CML image is picking from master

**Comment by 0x2b3bfa0** [comment]
> _however the CML image is picking from master <sup>[citation required]</sup>_

As far as I know, we don't do that anymore <sup>[[1](https://github.com/iterative/terraform-provider-iterative/blob/5f62a02d428eb9f6961e1f0721ec76364d7fe347/iterative/resource_runner.go#L275), [2](https://github.com/iterative/cml/blob/d716d3072d4d9829865d95f67bdf79cc7c9c5a62/Dockerfile#L78-L79)]</sup>

**Comment by 0x2b3bfa0** [comment]
> _testbot should do the trick however the CML image is picking from master_

Do you mean that the [checkbot](https://github.com/iterative/cml/blob/master/.github/workflows/checkbot.yml) test image is being built from the latest changes in the branch that triggered the workflow?

**Comment by DavidGOrtega** [comment]
No, what I say is that despite that we are testing everything when we launch a cloud runner it is going to use "live" cml not "dev", hence we are not going to notice those issues and testing with cloud runners is unsecure

**Comment by 0x2b3bfa0** [comment]
> [_As per #527, I would tend to avoid integration tests whenever possible and detect these issues by resorting to simpler unit tests._](https://github.com/iterative/cml/issues/550#issuecomment-845166152)

**Comment by 0x2b3bfa0** [comment]
### Suggested action points

- [ ] Use [custom setup and teardown](https://jestjs.io/docs/setup-teardown#one-time-setup) functions to create and destroy ephemeral test repositories on GitHub, GitLab and Bitbucket where we can run each test in a clean slate and without getting extra notifications. Most of these services can be executed inside GitHub Actions as [service containers](https://docs.github.com/es/actions/guides/about-service-containers#about-service-containers); see the [`gitlab.yml`](https://github.com/iterative/cml/blob/899c6d2e3eaffc66d12492deffa658e8430f1c66/.github/workflows/gitlab.yml
) workflow and our internal on-premise Bitbucket server [roles](https://github.com/iterative/viewer-itops/blob/bf9eec5ef300cabba57eb7f2adfa0d3a759357c2/ansible/onpremise/roles/bitbucket/tasks/main.yaml) from https://github.com/iterative/viewer/issues/1051.

- [ ] Create a minimal directory structure to provision these ephemeral repositories. It can be composable with overlays so we keep a single code base for all the different CI/CD syntax styles. See also [DavidGOrtega/fashion_mnist](https://github.com/DavidGOrtega/fashion_mnist) for an example of _whole picture integration tests_.

- [ ] Consider replacing [`checkbot.yml`](https://github.com/iterative/cml/blob/899c6d2e3eaffc66d12492deffa658e8430f1c66/.github/workflows/checkbot.yml) with acceptance tests that run on every pull request and don't include cloud functionality, which can be moved to the [iterative/terraform-provider-iterative](https://github.com/iterative/terraform-provider-iterative) repository.

### Already solved issues

- [X] https://github.com/iterative/cml/pull/639
- [X] https://github.com/iterative/cml/pull/633
- [X] https://github.com/iterative/cml/pull/634
- [X] https://github.com/iterative/terraform-provider-iterative/pull/111

**Comment by 0x2b3bfa0** [comment]
Not necessarily blocked by https://github.com/iterative/cml/issues/645 unless we plan to send GitHub checks between repositories. In fact, I would prefer to avoid that approach and resourt to our [`jest`](jestjs.io) tests instead to verify that the triggered actions have altered the target repository status in the expected way.
